src/ecmult_const_impl.h

```diff
@@ -152,6 +152,10 @@ static void secp256k1_ecmult_const(secp256k1_gej *r, const secp256k1_ge *a, cons
 
     /* build wnaf representation for q. */
     int rsize = size;
+    if (secp256k1_ge_is_infinity(a)) {
+      secp256k1_gej_set_infinity(r);
+      return;
+    }
```

Comment:
I think this fix can also be applied to `ecmult` for the `a` argument. Just replacing it with an `ecmult_gen` in the case that `a` is infinity. Otherwise the check here fails: https://github.com/bitcoin-core/secp256k1/blob/master/src/ecmult_impl.h#L102

Reply:
ecmult_gen isn't necessarily available to ecmult, as it requires a signing context and also would be a non-trivial code-size increase for something that was validation only. 

```c
        if (secp256k1_scalar_is_zero(&na[np]) || secp256k1_gej_is_infinity(&a[np])) {
            continue;
        }
```

should be skipping the input if its an infinity.

Reply:
Ah sorry I missed that. I was mistakenly projecting what happens in the rust parity code base onto this.